engine/src/main/java/com/redhat/cloud/notifications/processors/webhooks/WebhookTypeProcessor.java

```diff
@@ -254,8 +272,10 @@ public void doHttpRequest(Event event, Endpoint endpoint, HttpRequest<Buffer> re
                 }
 
                 if (serverError) {
+                    updateRetryMetrics(isEmailEndpoint);
```

Comment:
This only counts 1 of the 4 exceptions that can trigger a retry.

Reply:
FailSafe has an `onRetry` clause that could be used from the builder at the beginning of the class. The only problem is that we can't determine if we're retrying an email or a webhook call with the current retry policy. We could have two retry policies though and use the correct one depending on the endpoint type.